test/coverlet.core.tests/Instrumentation/InstrumenterTests.cs

```diff
@@ -367,18 +367,17 @@ public void SkipEmbeddedPpdbWithoutLocalSource()
         [Fact]
         public void SkipPpdbWithoutLocalSource()
         {
-            Mock<IFileSystem> mockFileSystem = new Mock<IFileSystem>();
-            mockFileSystem.Setup(fs => fs.Exists(It.IsAny<string>())).Returns((string path) =>
-             {
-                 return Path.GetExtension(path) != ".cs";
-             });
-            InstrumentationHelper instrumentationHelper = new InstrumentationHelper(new ProcessExitHandler(), new RetryHelper(), mockFileSystem.Object);
-
             string coverletLib = Directory.GetFiles(Directory.GetCurrentDirectory(), "coverlet.core.dll").First();
+            var instrumentationHelperMock = new Mock<IInstrumentationHelper>();
```

Comment:
Why you've changed this test?
Here I want to test `InstrumentationHelper` logic and I have to mock only file system to simulate "file not found".
Revert please.

Reply:
I don't exactly know what you mean or how I can make this test running again by only mocking the FileSystem. When I revert the test to how it was before the test fails. This is because I now also use the FileSystem adapter for the streams. Thus I also must setup e.g. ```_fileSystem.OpenRead()```.... 

And wouldn't it be better to test the ```InstrumentationHelper``` logic in the class ```InstrumentationHelperTests```.

Reply:
> And wouldn't it be better to test the InstrumentationHelper logic in the class InstrumentationHelperTests.

Here we have a bit of integration...I mean we need to test the interaction between InstrumentationHelper and Instrumenter.
You should do partial mock of concrete implementation and redefine only `Exists` something like 
https://stackoverflow.com/questions/4769928/using-moq-to-mock-only-some-methods
```
var mock = new Mock<YourTestClass>();
 mock.CallBase = true;
 mock.Setup(x => x.GetCurrentUser()).Returns(lUnauthorizedUser);
 mockedTest.Object.MyMethod();
```

Reply:
In this test we need to simulate to find pdb file but return false when we try to understand if sources are present on local hard drive, so we did it with `coverlet.core.dll` to avoid new dependency and we need to return false also if we have sources in local.

Reply:
Ok thanks for the information with the partial mock. :thumbsup: